app/javascript/components/key-performance-indicators/common-date-ranges.js

```diff
@@ -0,0 +1,45 @@
+import { subMonths, startOfMonth, endOfMonth } from "date-fns";
+
+import { DateRange } from "./date-range-select";
+
+// not sure how to use i18n outside of react yet.
+const i18n = window.I18n;
```

Comment:
I think I see why this isn't working. You are going to have to pass i18n from a component to the properties below. So

```
   AllTime: translate => new DateRange(
        "all-time",
        translate("key_performance_indicators.time_periods.all_time"),
        new Date(Date.parse("01/01/2000")),
        endOfMonth(today)
      ),

... In component
const i18n = useI18n()
....
CommonDateRanges.AllTime(i18n.t)
```

Reply:
Is this because the component is somehow rendered outside of the Provider for the I18n stuff?